### PR TITLE
Bump GMP dep and add msan to MPFR

### PR DESCRIPTION
**Pull request by gbaraldi**

### CODE REVIEW

File: M/MPFR/build_tarballs.jl [hunk 1]
```diff
@@ -34,7 +40,8 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("GMP_jll", v"6.2.0"),
+    Dependency("GMP_jll", v"6.2.1"),
+    BuildDependency("LLVMCompilerRT_jll",platforms=[Platform("x86_64", "linux"; sanitize="memory")]),
```
Comment (giordano) [inline M/MPFR/build_tarballs.jl:44]: I prefer to filter like this https://github.com/JuliaPackaging/Yggdrasil/blob/c8a9c2ae6022f6e679ae3a82153060b20d55c846/O/OpenBLAS/common.jl#L233

Comment (giordano) [reply M/MPFR/build_tarballs.jl:44]: Patter which, by the way, would have made #5458 actually work if you wanted to go by that route.

Comment (gbaraldi) [reply M/MPFR/build_tarballs.jl:44]: So it's more futureproof? But sure

Comment (giordano) [reply M/MPFR/build_tarballs.jl]: Yes, but also, matching platforms is very finicky, I don't generally trust it, filtering by properties is more accurate.

**Pull request merged**